File: src/01-easy/power-of-thor-episode-1/solution.ts
const inputs = readline().split(' ').map(Number);
let [lx, ly, tx, ty] = inputs;
while (true) {
    readline();
    let dir = '';
    if (ty > ly) { dir += 'N'; ty--; }
    else if (ty < ly) { dir += 'S'; ty++; }
    if (tx > lx) { dir += 'W'; tx--; }
    else if (tx < lx) { dir += 'E'; tx++; }
    console.log(dir);
}

File: src/01-easy/power-of-thor-episode-1/solution.swift
while true {
    let line = readLine()!.split(" ").map { Int($0)! }
    var lx = line[0], ly = line[1], tx = line[2], ty = line[3]
    while true {
        _ = readLine()
        var dir = ""
        if ty > ly { dir += "N"; ty -= 1 }
        else if ty < ly { dir += "S"; ty += 1 }
        if tx > lx { dir += "W"; tx -= 1 }
        else if tx < lx { dir += "E"; tx += 1 }
        print(dir)
    }
}
